### PR TITLE
bases: add almalinux support

### PR DESCRIPTION
**Pull request by syu-w**
CRAFT-1728

### CODE REVIEW

File: charmcraft/providers.py [hunk 1]
```diff
@@ -241,10 +241,11 @@ def is_base_available(base: Base) -> Tuple[bool, Union[str, None]]:
             f"host architecture {arch!r} not in base architectures {base.architectures!r}",
         )
 
-    if base.name not in ("ubuntu", "centos"):
+    if base.name not in ("ubuntu", "centos", "almalinux"):
```
[the remark this answers is not in the document]
Comment (syu-w) [reply charmcraft/providers.py:244]: Sounds good.

File: requirements-dev.txt [hunk 2]
```diff
@@ -9,8 +9,10 @@ click==8.1.3
 colorama==0.4.6
 coverage==7.2.3
 craft-cli==1.2.0
-craft-parts==1.19.3
-craft-providers==1.11.0
+#craft-parts==1.19.3
+git+https://github.com/syu-w/craft-parts.git@almalinux#craft-parts
+#craft-providers==1.11.0
+git+https://github.com/syu-w/craft-providers.git@almalinux#craft-providers
```
[the remark this answers is not in the document]
Comment (syu-w) [reply requirements-dev.txt:15]: Of course.

File: tests/spread/smoketests/basic-almalinux/task.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+summary: pack a simple init-created charm for AlmaLinux 9
+systems:
+  - ubuntu-22.04-64
+kill-timeout: 30m
```
Comment (lengau) [inline tests/spread/smoketests/basic-almalinux/task.yaml:4]: Do we need to increase the kill timeout from the default 15m to 30m here?

File: charmcraft/providers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (syu-w) [reply charmcraft/providers.py:244]: Maybe I should put this one in craft-providers?

**Review by sergiusens — APPROVED** [review]

File: charmcraft/providers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lengau) [reply charmcraft/providers.py:244]: `craft-providers` is a sensible long-term spot for it

**Pull request merged**